### PR TITLE
fix(markdown-code): preserve nested container boundaries

### LINKED ISSUE #1858: fix(markdown-code): preserve nested container boundaries
## Background

#1856 and its merged PR #1857 now mask Markdown inline and fenced code regions before the A4.5 Check 3 policy-pattern evaluation. A residual, reproducible boundary class remains on `origin/main` at `b5dcf356`: the position-preserving Markdown scanner can classify a policy phrase incorrectly when code spans cross an indented list boundary, when a fenced block is continued by a multi-digit list item or across a blank line, or when an inline span uses a lazy blockquote continuation.

These are narrow follow-ups in `src/scripts/markdown-code.mts`, not a change to the policy keyword families or to the other Trust/Safety scanners. The current main implementation returns `pass: true` for visible prose after the list-fence cases and `pass: false` for the policy phrase contained in a lazy blockquote span; the corresponding focused cases pass with the verified `3db67eee` implementation. The behavior is independently reproducible through `checkTrustSafety` on both revisions.

Refs #1856.

## Proposed change

- Refine the shared Markdown code-region range and masking logic so container boundaries are interpreted consistently for indented list items, multi-digit and blank-line list continuations, and lazy blockquote lines.
- Preserve the existing position-preserving contract and the distinction between inert Markdown code and ordinary issue prose.
- Keep the Trust/Safety policy-pattern scope narrow; do not alter secret detection, unsafe-execution detection, duplicate detection, or the policy keyword families.
- Add focused regressions for each boundary class and retain controls showing that visible policy prose still fails Check 3 while code-only examples remain masked.
- Regenerate generated helper output from the `.mts` source and inspect the diff for unrelated changes.

## Acceptance criteria

- [ ] A policy phrase in an inline span that crosses an indented list block boundary remains visible as prose when the boundary ends the span.
- [ ] A fenced block under a multi-digit ordered list masks only the fenced content and exposes the visible continuation after its properly indented closer.
- [ ] A list-item fenced block preserves its active list indentation across a blank line and exposes a continuation that is indented relative to that list item.
- [ ] An inline code span in a quoted paragraph remains masked across a lazy blockquote continuation line.
- [ ] Existing nested blockquote, list-item fence, invalid-fence, indentation, position-preservation, and ordinary-prose Trust/Safety regressions continue to pass.
- [ ] The change stays limited to Markdown range/masking behavior and the Check 3 regression fixtures; secret and unsafe-execution checks remain unchanged.
- [ ] `scripts/markdown-code.mjs` and any affected generated helper are regenerated from their `.mts` sources with no unrelated drift.
- [ ] `node --test tests/markdown-code.test.mts tests/suitability-triage.test.mts` and the repository lint/build checks pass.

---

_Autopilot suitability: 5 / 5 -- higher is more autopilot-suitable; below the configured floor is human-oriented._

_Effort: M -- author-estimated size; a soft autopilot selection tie-breaker only._

### PR DESCRIPTION
**Pull request by kurone-kito**
# Summary

Preserve Markdown code masking across nested list and blockquote container boundaries. The change closes the residual false-positive/false-negative cases found after #1857 without changing Check 3 policy keywords or the other Trust/Safety scanners.

## Linked issue

Closes #1858

## Follow-up issues (if any)

- [x] none

## Background / rationale (if material)

`origin/main` still treated a policy phrase incorrectly when an inline span crossed an indented list boundary, when list-item fences used multi-digit or blank-line continuations, or when a quoted inline span continued lazily onto the next line. The shared position-preserving Markdown range scanner now tracks those container boundaries consistently.

## IDD impact

- [ ] Instruction files changed
- [ ] Template files changed
- [x] Helper scripts changed
- [ ] Config schema changed
- [x] Security / credential / merge behavior changed

## Verification

- [x] `pnpm lint`
- [ ] `pnpm test` (covered by the full test run inside `pnpm lint`)
- [x] `node scripts/audit-docs.mjs --check`
- [ ] `pnpm run docs:sync:check`
- [x] `node scripts/idd-doctor.mjs` (if applicable)

## Safety

- [x] No secret-bearing examples
- [x] No private repo names / URLs
- [x] Merge policy impact reviewed
- [x] Context budget impact reviewed

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Markdown code-block detection within nested lists and blockquotes.
  * Correctly handles indentation, fenced code, lazy quote continuations, HTML blocks, and list boundaries.
  * Prevents code examples from being incorrectly masked or merged across Markdown structures.
  * Improved overlap scanning to avoid premature termination and false matches.

* **Tests**
  * Added regression coverage for nested lists, quoted fences, indentation, HTML interruptions, and scanning edge cases.

### CODE REVIEW

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 4228aeb64269d1540a17dc373f5e3edb9f17f394 2026-08-04T16:48:46Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 497f9fd4640824e68ef908dc734ba3dab7264da7 2026-08-04T17:06:29Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

**Comment by kurone-kito** [comment]
| Field | Value |
| --- | --- |
| Phase | E14 advisory wait |
| Claim | 6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 |
| Branch | issue/1858-fix-markdown-code-preserve-nested |
| Last checked | 2026-08-04T17:07:23.255Z |
| Open blockers | Copilot review for HEAD 497f9fd4; advisory run 30932317572 is pending fresh review; review/disposition triage follows |
| Next action | Poll advisory-wait-state, then rerun the failed advisory job once after same-HEAD Copilot review |
| Authoritative by | advisory-wait-recovery 5182238204; current HEAD 497f9fd4; pnpm lint |

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 72d3b9dee05d5ba083e4ffbc53241a15c2304060 2026-08-04T17:23:57Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 921c72e8964ccad306b09bf728429d37fba5cc9e 2026-08-04T17:37:34Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

File: src/scripts/markdown-code.mts [hunk 1]
```diff
@@ -158,14 +158,25 @@ function findMarkdownBlockBoundary(
...
+      MARKDOWN_BLOCK_CONTENT_PATTERN.test(parsed.content) ||
+      (fencedLine !== null && isValidFenceOpener(fencedLine));
     if (parsed.containerDepth !== openingContainerDepth) {
       // A quote marker may continue an inline span only when it belongs to
       // the same container. A quote that starts or ends here is a block break.
-      if (parsed.containerDepth > 0 || openingContainerDepth > 0) {
+      const isLazyQuoteContinuation =
+        openingContainerDepth > 0 &&
+        parsed.containerDepth === 0 &&
+        !isBlockStart;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:191]: **Accepted** — fixed in commits 497f9fd4 and 72d3b9de: HTML block boundaries now terminate the quoted inline-span scan.

File: src/scripts/markdown-code.mts [hunk 2]
```diff
@@ -410,6 +473,15 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
         return fencedLine !== null && isValidFenceOpener(fencedLine);
       })();
     previousContainerDepth = parsed.containerDepth;
+    if (parsed.listContentIndent !== null) {
+      activeListContentIndent = parsed.listContentIndent;
+    } else if (
+      !isBlank &&
+      activeListContentIndent !== null &&
+      indentationColumns(parsed.content) < activeListContentIndent
+    ) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 497f9fd4: two blank lines now end the active list container before the following code block.

File: src/scripts/markdown-code.mts [hunk 3]
```diff
@@ -275,7 +330,11 @@ function parseFencedLine(line: string): FencedLine | null {
   // A fenced block may begin directly after a list marker (`- ~~~` or
   // `1. ~~~`). The list marker is a container prefix, not part of the fence;
   // continuation lines commonly carry only the list indentation (`  ~~~`).
-  const content = stripListItemMarker(containerContent);
+  const relativeContent =
+    activeListContentIndent === null
+      ? containerContent
+      : stripLeadingIndentColumns(containerContent, activeListContentIndent);
+  const content = stripListItemMarker(relativeContent);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commits 497f9fd4 and 72d3b9de: open fence content is kept opaque and list markers are not reparsed.

File: src/scripts/markdown-code.mts [hunk 4]
```diff
@@ -410,6 +473,15 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
         return fencedLine !== null && isValidFenceOpener(fencedLine);
       })();
     previousContainerDepth = parsed.containerDepth;
+    if (parsed.listContentIndent !== null) {
+      activeListContentIndent = parsed.listContentIndent;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 497f9fd4: over-wide list-marker padding is normalized using CommonMark continuation semantics.

[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 497f9fd4: indented-code content no longer updates list state.

File: src/scripts/markdown-code.mts [hunk 5]
```diff
@@ -374,13 +434,16 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
...
   let lineStart = 0;
 
   while (lineStart <= text.length) {
     const line = lineBounds(text, lineStart);
     const rawLine = text.slice(lineStart, line.end);
     const parsed = parseContainerLine(rawLine);
-    const isIndented = indentationColumns(parsed.content) >= 4;
+    const isIndented =
+      indentationColumns(parsed.content) >=
+      (activeListContentIndent === null ? 4 : activeListContentIndent + 4);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:548]: **Accepted** — fixed in commit 497f9fd4: list indentation is reset when blockquote depth changes.

File: src/scripts/markdown-code.mts [hunk 6]
```diff
@@ -275,7 +330,11 @@ function parseFencedLine(line: string): FencedLine | null {
   // A fenced block may begin directly after a list marker (`- ~~~` or
   // `1. ~~~`). The list marker is a container prefix, not part of the fence;
   // continuation lines commonly carry only the list indentation (`  ~~~`).
-  const content = stripListItemMarker(containerContent);
+  const relativeContent =
+    activeListContentIndent === null
+      ? containerContent
+      : stripLeadingIndentColumns(containerContent, activeListContentIndent);
+  const content = stripListItemMarker(relativeContent);
   const fenceMatch = content.match(/^ {0,3}(`{3,}|~{3,})(.*)$/u);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:398]: **Accepted** — fixed in commits 497f9fd4 and 72d3b9de: fenced list content is no longer mistaken for a fence marker.

File: scripts/markdown-code.mjs [hunk 7]
```diff
@@ -217,7 +265,11 @@ function parseFencedLine(line) {
   // A fenced block may begin directly after a list marker (`- ~~~` or
   // `1. ~~~`). The list marker is a container prefix, not part of the fence;
   // continuation lines commonly carry only the list indentation (`  ~~~`).
-  const content = stripListItemMarker(containerContent);
+  const relativeContent =
+    activeListContentIndent === null
+      ? containerContent
+      : stripLeadingIndentColumns(containerContent, activeListContentIndent);
+  const content = stripListItemMarker(relativeContent);
   const fenceMatch = content.match(/^ {0,3}(`{3,}|~{3,})(.*)$/u);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply scripts/markdown-code.mjs:335]: **Accepted** — fixed in commits 497f9fd4 and 72d3b9de: generated output now preserves the same opaque fence-content handling.

File: src/scripts/markdown-code.mts [hunk 8]
```diff
@@ -119,9 +119,11 @@ function hasBlankLine(text: string, start: number, end: number): boolean {
 }
 
 const MARKDOWN_BLOCK_CONTENT_PATTERN =
-  /^(?:#{1,6}(?:[ \t]|$)|(?:[-+*])[ \t]+|1[.)][ \t]+|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
+  /^ {0,3}(?:#{1,6}(?:[ \t]|$)|(?:[-+*])[ \t]+|1[.)][ \t]+|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
 const MARKDOWN_INDENTED_CODE_PRECEDER_PATTERN =
-  /^(?:#{1,6}(?:[ \t]|$)|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
+  /^ {0,3}(?:#{1,6}(?:[ \t]|$)|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:128]: **Rejected** — verified against CommonMark paragraph interruption semantics and the existing regression test for non-one ordered markers inside an inline code span: only an ordered marker starting with 1 may interrupt a paragraph. Broadening MARKDOWN_BLOCK_CONTENT_PATTERN to 10. or 2) would regress that behavior, so the intentional 1[.)] boundary is retained.

File: src/scripts/markdown-code.mts [hunk 9]
```diff
@@ -119,9 +119,11 @@ function hasBlankLine(text: string, start: number, end: number): boolean {
...
+const MARKDOWN_HTML_BLOCK_START_PATTERN =
+  /^ {0,3}(?:<!--|<\?|<![A-Z]|<!\[CDATA\[|<\/?(?:address|article|aside|base|basefont|blockquote|body|caption|center|col|colgroup|dd|details|dialog|dir|div|dl|dt|fieldset|figcaption|figure|footer|form|h[1-6]|head|header|hr|html|iframe|legend|li|link|main|menu|menuitem|nav|ol|p|pre|script|section|summary|table|tbody|td|tfoot|th|thead|title|tr|track|ul)(?:[ \t/>]|$))/u;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 921c72e8: interrupting HTML block tag matching is now case-insensitive, including style and textarea.

File: src/scripts/markdown-code.mts [hunk 10]
```diff
@@ -410,6 +509,19 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
         return fencedLine !== null && isValidFenceOpener(fencedLine);
       })();
     previousContainerDepth = parsed.containerDepth;
+    if (rangeStart === null && parsed.listContentIndent !== null) {
+      activeListContentIndent = parsed.listContentIndent;
+      activeListContainerDepth = parsed.containerDepth;
+      activeListBlankLines = 0;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 921c72e8: non-one ordered markers following an ordinary paragraph no longer establish list indentation state.

File: src/scripts/markdown-code.mts [hunk 11]
```diff
@@ -374,14 +449,38 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
...
+      activeListContentIndent !== null &&
+      parsed.containerDepth !== activeListContainerDepth
+    ) {
+      activeListContentIndent = null;
+      activeListContainerDepth = null;
+      activeListBlankLines = 0;
+    }
+    const isIndented =
+      indentationColumns(parsed.content) >=
+      (activeListContentIndent === null ? 4 : activeListContentIndent + 4);
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:548]: **Accepted** — fixed in commit 921c72e8: tabbed list padding is expanded from the marker column.

File: src/scripts/markdown-code.mts [hunk 12]
```diff
@@ -158,14 +160,26 @@ function findMarkdownBlockBoundary(
...
+      MARKDOWN_HTML_BLOCK_START_PATTERN.test(parsed.content) ||
+      (fencedLine !== null && isValidFenceOpener(fencedLine));
     if (parsed.containerDepth !== openingContainerDepth) {
       // A quote marker may continue an inline span only when it belongs to
       // the same container. A quote that starts or ends here is a block break.
-      if (parsed.containerDepth > 0 || openingContainerDepth > 0) {
+      const isLazyQuoteContinuation =
+        openingContainerDepth > 0 &&
+        parsed.containerDepth === 0 &&
+        !isBlockStart;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:191]: **Accepted** — fixed in commit 921c72e8: lazy quote continuation is limited to paragraph content, so a quoted heading ends the inline span.

File: src/scripts/markdown-code.mts (hunk 10 above)
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Rejected** — the proposed fixture opens a five-backtick fence but closes it with three backticks; CommonMark requires the closing fence to be at least as long as the opener. The current implementation correctly keeps the policy text masked, so this is not a valid regression.

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD 921c72e8964ccad306b09bf728429d37fba5cc9e (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5181993799)

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 3f5075834a9ae71c21c4c1f0be2ad53a18aab654 2026-08-04T17:55:08Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 20cb73205833a6962a96eed75bc5040639e35cc5 2026-08-04T18:10:32Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

**Comment by kurone-kito** [comment]
**Accepted** — coderabbitai[bot] summary walkthrough at HEAD 20cb73205833a6962a96eed75bc5040639e35cc5; actionable comments, if any, are dispositioned as their own review threads

File: src/scripts/markdown-code.mts [hunk 13]
```diff
@@ -119,9 +123,11 @@ function hasBlankLine(text: string, start: number, end: number): boolean {
 }
 
 const MARKDOWN_BLOCK_CONTENT_PATTERN =
-  /^(?:#{1,6}(?:[ \t]|$)|(?:[-+*])[ \t]+|1[.)][ \t]+|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
+  /^ {0,3}(?:#{1,6}(?:[ \t]|$)|(?:[-+*])[ \t]+|1[.)][ \t]+|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
 const MARKDOWN_INDENTED_CODE_PRECEDER_PATTERN =
-  /^(?:#{1,6}(?:[ \t]|$)|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
+  /^ {0,3}(?:#{1,6}(?:[ \t]|$)|(?:-{1,}|={1,}|_{3,}|\*{3,})[ \t]*$)/u;
+const MARKDOWN_HTML_BLOCK_START_PATTERN =
+  /^ {0,3}(?:<!--|<\?|<![A-Z]|<!\[CDATA\[|<\/?(?:address|article|aside|base|basefont|blockquote|body|caption|center|col|colgroup|dd|details|dialog|dir|div|dl|dt|fieldset|figcaption|figure|footer|form|h[1-6]|head|header|hr|html|iframe|legend|li|link|main|menu|menuitem|nav|ol|p|pre|script|section|style|summary|table|tbody|td|textarea|tfoot|th|thead|title|tr|track|ul)(?:[ \t/>]|$))/u;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 3f507583: HTML block tags now require a `>` or `/>` terminator, so a bare slash remains paragraph text.

File: src/scripts/markdown-code.mts [hunk 14]
```diff
@@ -238,19 +256,43 @@ function parseListItemContainer(content: string): number | null {
...
+    contentPadding
   );
 }
 
 function parseContainerLine(line: string): ContainerLine {
-  const quoteMatch = line.match(/^ {0,3}(?:(?:> ?)+)(.*)$/u);
-  const content = quoteMatch ? quoteMatch[1] : line;
+  let cursor = 0;
+  let containerDepth = 0;
+  while (cursor < line.length) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:287]: **Accepted** — fixed in commit 20cb7320: fenced list content now preserves outer list and blockquote boundaries without reparsing quote-like lines.

File: src/scripts/markdown-code.mts [hunk 15]
```diff
@@ -238,19 +265,42 @@ function parseListItemContainer(content: string): number | null {
...
-  );
+  const markerEndColumns =
+    indentationColumns(listItem.markerIndent) + listItem.marker.length;
+  const spacingColumns =
+    indentationColumns(listItem.spacing, markerEndColumns) - markerEndColumns;
+  // CommonMark treats five or more spaces after a list marker as one
+  // separating space plus literal content indentation. Keeping the full
+  // padding here would make a valid four-column continuation look like
+  // ordinary prose instead of nested code.
+  const contentPadding = spacingColumns > 4 ? 1 : spacingColumns;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:280]: **Accepted** — fixed in commit 20cb7320: over-wide list padding is preserved as literal indentation instead of opening a fence.

File: src/scripts/markdown-code.mts [hunk 16]
```diff
@@ -410,6 +517,19 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
...
     previousContainerDepth = parsed.containerDepth;
+    if (rangeStart === null && parsed.listContentIndent !== null) {
+      activeListContentIndent = parsed.listContentIndent;
+      activeListContainerDepth = parsed.containerDepth;
+      activeListBlankLines = 0;
+    } else if (
+      !isBlank &&
+      activeListContentIndent !== null &&
+      indentationColumns(parsed.content) < activeListContentIndent
+    ) {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — fixed in commit 3f507583: deindented lines clear list state before the code-indentation threshold is calculated.

File: src/scripts/markdown-code.mts [hunk 17]
```diff
@@ -150,22 +156,43 @@ function findMarkdownBlockBoundary(
...
+  const openingRawLine = text.slice(openingLineStart, openingLine.end);
+  const openingParsed = parseContainerLine(openingRawLine);
+  const openingListItem = parseListItemMatch(openingParsed.content);
+  const openingParagraphContent =
+    openingListItem?.content ?? openingParsed.content;
+  const openingFence = parseFencedLine(openingRawLine);
+  const openingIsParagraph =
+    !MARKDOWN_BLOCK_CONTENT_PATTERN.test(openingParagraphContent) &&
+    !MARKDOWN_HTML_BLOCK_START_PATTERN.test(openingParagraphContent) &&
+    (openingFence === null || !isValidFenceOpener(openingFence));
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:171]: **Accepted** — fixed in commit 20cb7320: custom HTML block openers now prevent invalid lazy quote continuation.

**Comment by kurone-kito** [comment]
advisory-wait-recovery: codex-848483b6-cabc-48e4-8d2c-77182322ec9d 1fa70784e54d1b7b8df4b0b9ce33d471bb4e87ae 2026-08-04T18:24:11Z claim:6a57bf07-dc0c-40ab-bcca-2b1ed4dfea10 attempt:1

**Comment by kurone-kito** [comment]
_codex-848483b6-cabc-48e4-8d2c-77182322ec9d: review triage snapshot — IDD automation marker. Do not edit._

File: src/scripts/markdown-code.mts [hunk 18]
```diff
@@ -150,22 +158,45 @@ function findMarkdownBlockBoundary(
...
+  const openingParsed = parseContainerLine(openingRawLine);
+  const openingListItem = parseListItemMatch(openingParsed.content);
+  const openingParagraphContent =
+    openingListItem?.content ?? openingParsed.content;
+  const openingFence = parseFencedLine(openingRawLine);
+  const openingIsParagraph =
+    !MARKDOWN_BLOCK_CONTENT_PATTERN.test(openingParagraphContent) &&
+    !MARKDOWN_HTML_BLOCK_START_PATTERN.test(openingParagraphContent) &&
+    !MARKDOWN_CUSTOM_HTML_BLOCK_START_PATTERN.test(openingParagraphContent) &&
+    (openingFence === null || !isValidFenceOpener(openingFence));
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:171]: Accepted: follow-up tracked in #1862; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 19]
```diff
@@ -150,22 +158,45 @@ function findMarkdownBlockBoundary(
...
+      (fencedLine !== null && isValidFenceOpener(fencedLine));
     if (parsed.containerDepth !== openingContainerDepth) {
       // A quote marker may continue an inline span only when it belongs to
       // the same container. A quote that starts or ends here is a block break.
-      if (parsed.containerDepth > 0 || openingContainerDepth > 0) {
+      const isLazyQuoteContinuation =
+        openingContainerDepth > 0 &&
+        parsed.containerDepth === 0 &&
+        openingIsParagraph &&
+        !isBlockStart;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:191]: Accepted: follow-up tracked in #1862; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 20]
```diff
@@ -150,22 +158,45 @@ function findMarkdownBlockBoundary(
...
+      MARKDOWN_HTML_BLOCK_START_PATTERN.test(parsed.content) ||
+      MARKDOWN_CUSTOM_HTML_BLOCK_START_PATTERN.test(parsed.content) ||
+      (fencedLine !== null && isValidFenceOpener(fencedLine));
     if (parsed.containerDepth !== openingContainerDepth) {
       // A quote marker may continue an inline span only when it belongs to
       // the same container. A quote that starts or ends here is a block break.
-      if (parsed.containerDepth > 0 || openingContainerDepth > 0) {
+      const isLazyQuoteContinuation =
+        openingContainerDepth > 0 &&
+        parsed.containerDepth === 0 &&
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:189]: Accepted: follow-up tracked in #1862; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 21]
```diff
@@ -410,6 +585,11 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
         return fencedLine !== null && isValidFenceOpener(fencedLine);
       })();
     previousContainerDepth = parsed.containerDepth;
+    if (rangeStart === null && listContentIndent !== null) {
+      activeListContentIndent = listContentIndent;
+      activeListContainerDepth = parsed.containerDepth;
+      activeListBlankLines = 0;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:591]: Accepted: follow-up tracked in #1863; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 22]
```diff
@@ -238,25 +269,77 @@ function parseListItemContainer(content: string): number | null {
...
+      cursor = markerStart;
+      break;
+    }
+    cursor += 1;
+    containerDepth += 1;
+    if (line[cursor] === ' ') {
+      cursor += 1;
+    }
+  }
+  const content = containerDepth > 0 ? line.slice(cursor) : line;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:304]: Accepted: follow-up tracked in #1863; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 23]
```diff
@@ -266,7 +349,34 @@ function continuesListContainer(
...
+  }
+  let columns = 0;
+  let cursor = 0;
+  while (cursor < text.length && columns < targetColumns) {
+    const character = text[cursor];
+    if (character === ' ') {
+      columns += 1;
+    } else if (character === '\t') {
+      columns += 4 - (columns % 4);
+    } else {
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:367]: Accepted: follow-up tracked in #1863; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 24]
```diff
@@ -374,14 +503,60 @@ function findIndentedCodeRanges(text: string): MarkdownCodeRange[] {
...
+      indentationColumns(parsed.content) < activeListContentIndent
+    ) {
+      activeListContentIndent = null;
+      activeListContainerDepth = null;
+      activeListBlankLines = 0;
+    }
+    const isNonInterruptingOrderedItem: boolean =
+      listItem !== null &&
+      /^\d{1,9}[.)]$/u.test(listItem.marker) &&
+      !/^1[.)]$/u.test(listItem.marker) &&
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:538]: Accepted: follow-up tracked in #1863; the issue remains under the authoring hold until release.

File: src/scripts/markdown-code.mts [hunk 25]
```diff
@@ -238,25 +269,77 @@ function parseListItemContainer(content: string): number | null {
...
-  return parseListItemMatch(content)?.content ?? content;
+  const listItem = parseListItemMatch(content);
+  if (!listItem) {
+    return content;
+  }
+  const markerEndColumns =
+    indentationColumns(listItem.markerIndent) + listItem.marker.length;
+  const spacingColumns =
+    indentationColumns(listItem.spacing, markerEndColumns) - markerEndColumns;
+  const literalSpacing = spacingColumns > 4 ? listItem.spacing.slice(1) : '';
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts:341]: Accepted: follow-up tracked in #1863; the issue remains under the authoring hold until release.

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD 1fa70784e54d1b7b8df4b0b9ce33d471bb4e87ae (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5181993799)
